src/lib/Libcmds/pbs_json.c

```diff
@@ -152,6 +154,12 @@ add_json_node(JsonNodeType ntype, JsonValueType vtype, char *key, void *value) {
 				default:
 					buf[i++] = *ptr++;
 				}
+				if(i >= len-1) {
+					len += MAX_SPECIALCHAR_LEN;
+					buf = (char *)realloc(buf, len);
+					if(buf == NULL)
+						return NULL;
```

Comment:
Since the function is going into a library you might also want to free up buf here if realloc fails. And, also free it up in all other places where you want to bail out of the function.